source/slang/slang-emit-metal.cpp

```diff
@@ -653,6 +663,23 @@ void MetalSourceEmitter::emitSimpleTypeImpl(IRType* type)
             m_writer->emit("mesh_grid_properties ");
             return;
         }
+        case kIROp_MetalMeshType:
+        {
+            auto meshType = cast<IRMetalMeshType>(type);
+            m_writer->emit("metal::mesh<");
+            emitType(meshType->getVertexType());
+            m_writer->emit(", ");
+            emitType(meshType->getPrimitiveType());
+            m_writer->emit(", ");
+            emitSimpleValue(meshType->getMaxNumVertices());
+            m_writer->emit(", ");
+            emitSimpleValue(meshType->getMaxNumPrimitives());
+            m_writer->emit(", metal::topology::");
+            // TODO: this is pretty hacky, there has to be a better way to do this, but I dont see why this wouldn't work
+            m_writer->emitRawTextSpan(meshType->getTopology()->getStringSlice().begin(), meshType->getTopology()->getStringSlice().end());
```

Comment:
This can work, but from IR design point of view, it is better to have a enum value for the topology argument, and just use IRIntLit to represent topology. Then we can use an explicit switch statement here to emit the right text.

Reply:
Will do that

Reply:
how do i create a IRIntLit? there doesnt seem to be a builder emit function, and creating something so basic seems wrong

Reply:
You can use `IRBuilder::getIntValue(type, intVal)` to get/construct a integer literal IR inst from an int.